### DISCUSSION TITLE
ToiBot: A Simple Chatbot for Exploring the Flowise SDK

### DISCUSSION THREAD

**Discussion by toi500** [FlowiseAI/Flowise#3370] [Show and tell]
Hi people,
I built this chatbot so new Flowise users, including those with low-code or no-code experience, can easily start experimenting with the new Flowise SDK. It is a simple but effective starting point.
## Features
- **Streaming**: Get responses in a continuous stream, making conversations feel more natural and engaging.
- **Follow-up Prompts**:  Get suggestions for follow-up questions to keep users engaged.
- **Session Management**:  Context is maintained throughout your interactions thanks to session IDs.
- **Responsive Design**: Works great on any screen size.
- **Markdown Support**: Responses look great and are easy to read thanks to Markdown formatting.
- **Security**: The Flowise API key is used to make sure the chatbot is always secure.

https://github.com/user-attachments/assets/62676708-778e-42e9-8224-a88c015a73e6

Repo: https://github.com/toi500/toibot

**Comment by BlissiraHQ** [comment]
Amazing!!! Thanks so much for bringing this to the community! I didn't even know there was an SDK!! 

Edit: I get a stream error when running this. Would absolutely love to get to work, its an absolute game changer, you should be getting massive plaudits for this work! .. Any help massively appreciated! 

Stream error: SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON
!DOCTYPE "... is not valid JSON
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (node:internal/deps/undici/undici:5472:19)
    at successSteps (node:internal/deps/undici/undici:5454:27)
    at fullyReadBody (node:internal/deps/undici/undici:4381:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async consumeBody (node:internal/deps/undici/undici:5463:7)
    at async FlowiseClient.createPrediction (C:\Users\kushatron\Desktop\toibot-main\node_modules\flowise-sdk\dist\flowise-sdk.js:19:39)
    at async C:\Users\kushatron\Desktop\toibot-main\server.js:109:24

**Comment by toi500** [comment]
What chat model and Flowise version are you using? I just redeployed and works for me.

**Comment by BlissiraHQ** [comment]
Ah thanks so much for the reply, really appreciate it,  im running FW 2.2.2 ... Tried it with a number of my different flows.  It seems to trying to pass <!DOCTYPE through JSON, which obviously its failing as its not JSON safe. I can't for the life of me work out why its attempting such foul moves!

**Comment by BlissiraHQ** [comment]
I fixed this issue, schoolboy error. I had the full URL including my chatflow ID in the base path.  

Want to say a MASSIVE thanks for creating this, it's literally the most useful and game changing add on anyone has ever created for Flowise.!! The possibilities for this are endless, the apps I have created within just a few hours of tinkering are mind blowing! 

You are an absolute fkn LEGEND!

Can't thank you enough for this!!!

**Reply by toi500** [reply]
Thanks, man haha. I really appreciate your words. Happy to see you find it useful.

Since I released toibot I have also improved the official embed chatbot by making it deployable as a standalone app and adding custom CSS property for on-the-fly modifications.

https://github.com/FlowiseAI/FlowiseChatEmbed?tab=readme-ov-file#experimental-proxy-server-setup
https://docs.flowiseai.com/using-flowise/embed#custom-css-modification

It works really well too.
